src/compiler/graph/link_nodes.rs

```
@@ -0,0 +1,467 @@
+// TODO: remove this
+#![allow(unused, dead_code)]
+use paste::paste;
+
+use crate::types::{
+    block::{redstone::Component, Block, Facing, Kind},
+    compiler::{Graph, Link},
+    contraption::{Position, World},
+};
+
+pub fn link_nodes<'a, W: World<'a>>(mut graph: Graph, world: &'a W) -> Graph {
+    let indices = graph.node_indices();
+    let mut links = vec![];
+    for index in indices {
+        let node = graph.node_weight(index).expect("node should exist");
+        let block = world.get_block(node.pos).expect("block should exist");
```

Should return a error instead

---

fix this in new pr